code/modules/projectiles/pins.dm

```diff
@@ -112,6 +112,32 @@
 	icon_state = "firing_pin_pindi"
 	req_implant = /obj/item/weapon/implant/weapons_auth
 
+/obj/item/device/firing_pin/area
+	name = "area-denial pin"
+	desc = "this doesnt matter"
+	fail_message = "<span class='warning'>AREA CHECK FAILED.</span>"
+	var/area/A = null
+
+/obj/item/device/firing_pin/area/pin_auth(mob/living/user)
+	if(istype(get_area(src), A))
+		return 0
+	return 1
+
+/obj/item/device/firing_pin/area/syndicate
+	force_replace = 1
+	A = /area/syndicate_mothership
+	name = "Syndicate firing pin"
+	desc = "This pin will prevent firing whilst at the operative outpost."
+	icon_state = "firing_pin_pindi"
+
+/obj/item/device/firing_pin/trick
+	name = "Generic firing pin"
+	desc = "This pin will detonate the weapon it is put into upon trying to use it"
```

Comment:
It's saying it detonates the weapon when it is fired with the pin in - bad wording on my part.